initiateAlerter.cpp

```diff
@@ -0,0 +1,26 @@
+#include "initiateAlerter.h"
+#include "inputConfig.h"
+
+Alerter alerters[numberOfInputAlerts];
+
+Alerter *initiateAlerters()
+{
+    for (int i = 0; i < numberOfInputAlerts; i++)
+    {
+        AlertInterface *alrt;
+        switch (inputAlertsConfig[i].alertId)
+        {
+        case 0:
```

Comment:
This switch-case will keep growing as the variety of alertinterfaces increase.
Instead of switch-case, try an array of AlertInterface* 
0th element will be pointing to object of type AlertWithSMS, 1st to AlertWithSound... Later you can convert this into a map container (or use a map rightaway if you are familiar and with containers)

Reply:
> This switch-case will keep growing as the variety of alertinterfaces increase.
> Instead of switch-case, try an array of AlertInterface*
> 0th element will be pointing to object of type AlertWithSMS, 1st to AlertWithSound... Later you can convert this into a map container (or use a map rightaway if you are familiar and with containers)

In this project, we should not take predefined alerter's combination (like SMS and SOUND or only SMS etc).

Let's say, we have 5 (1, 2, 3, 4, 5) alert systems, each has it's own class and all inherits AlertInterface.
Now It's our consumer choice to choose his/her own alerter combination (like he/she choose 1, 3, 5) and i am taking these inputs in inputConfig file.
Now on the fly, I have to make an array of AlertInterface* having the addresses of objects of classes 1, 3 and 5 only.

I am using switch case because we have different classes for each alert system, so we have to switch between these as per the consumer choice on alerter's combination.

Is there any way to resolve switch-case in this scenerio while not comprising consumer freedom on alerter's combination.

Reply:
The switch-case says "0 means AlertWithSMS, 1 means AlertWithSound". This can also be done with a map, using which the number 0 can be mapped to a pointer, pointing to AlertWithSMS, and 1 can be mapped to a pointer to AlertWithSound.
If you are not familiar with maps, you can also think of an array like I wrote above, with 0th pointer in the array pointing to AlertWithSMS and 1st pointer pointing to AlertWithSound. It performs the same function as the switch-case with lesser code. Why does it compromise the consumer's freedom?

Reply:
> The switch-case says "0 means AlertWithSMS, 1 means AlertWithSound". This can also be done with a map, using which the number 0 can be mapped to a pointer, pointing to AlertWithSMS, and 1 can be mapped to a pointer to AlertWithSound.
> If you are not familiar with maps, you can also think of an array like I wrote above, with 0th pointer in the array pointing to AlertWithSMS and 1st pointer pointing to AlertWithSound. It performs the same function as the switch-case with lesser code. Why does it compromise the consumer's freedom?

sorry, I didn't get it at first place but now it is clear to me.
I have removed switch case by implementing the suggested method and now it also passes all the workflows.

Thanks,
Sudeep